source/tests/system/nidaqmx_driver_api_tests.cpp

```diff
@@ -922,6 +922,21 @@ TEST_F(NiDAQmxDriverApiTests, SetPreScaledUnits_GetPreScaledUnits_ReturnsAttribu
   EXPECT_EQ(ScaleInt32AttributeValues::SCALE_INT32_UNITS_PRE_SCALED_RPM, response.value());
 }
 
+TEST_F(NiDAQmxDriverApiTests, GetScaledUnitsAsDouble_Fails)
```

Comment:
Unfortunately, the trick of using the size=0 as a defensive mechanism does not work as well as I'd hoped. It works for getters using the wrong type, but not for setters.

So I think we will need a different way to handle "wrong type" accesses. Part of that will be enums that are organized by type. We can also generate a attribute-to-type validator to double-check. If we're going to allow for pure "attributes we don't know about", we might leave that behind a toggle. Or allow for plugging into the validator from the config file.

I'm slightly leaning towards keeping the hardcoded-zero-size-param. It's doing some good and provides a nice rule-of-3 third use case for the hardcoded_value mechanism.

Reply:
Ah, I see. I agree though, I'd lean towards keeping it as well. Like you said it's doing some good on its own plus gives some more validity to having the hardcoded_value mechanism.